R/build_stan_data.R

```diff
@@ -0,0 +1,245 @@
+#' for which FOI values will be estimated in [fit_seromodel].
+#' The max value in `foi_index`  correspond to the number of FOI values to
+#' be estimated when sampling.
+#' @inheritParams fit_seromodel
+#' @param group_size Age groups size
+#' @return Integer vector with the indexes numerating each year/age
+#' (depending on the model).
+#' @examples
+#' data(chagas2012)
+#' foi_index <- get_foi_index(chagas2012, group_size = 25)
+#' @export
+get_foi_index <- function(
+  serosurvey,
+  group_size
+  ) {
+    checkmate::assert_int(
+      group_size,
+      lower = 1,
+      upper = max(serosurvey$age_max)
+      )
+
+    foi_index <- unlist(
+      purrr::map(
+        seq(
+          1,
+          max(serosurvey$age_max) / group_size,
+          1),
+        rep,
+        times = group_size
+      )
+    )
+
+    foi_index <- c(
+      foi_index,
+      rep(
+        max(foi_index),
+        max(serosurvey$age_max) - length(foi_index)
+      )
+    )
+
+  return(foi_index)
+}
+
+#' Set stan data defaults for sampling
+#'
+#' @param stan_data List to be passed to [rstan][rstan::sampling]
+#' @inheritParams fit_seromodel
+#' @return List with default values of stan data for sampling
+set_stan_data_defaults <- function(
+    stan_data,
+    is_log_foi = FALSE,
+    is_seroreversion = FALSE
+) {
+  config_file <- system.file("extdata", "config.yml", package = "serofoi")
+  prior_default <- config::get(file = config_file, "priors")$defaults
+
+  foi_defaults <- list(
+    foi_prior_index = prior_default$index,
+    foi_min = prior_default$min,
+    foi_max = prior_default$max,
+    foi_mean = prior_default$mean,
+    foi_sd = prior_default$sd
+  )
+  # Add sigma defaults depending on scale
+  if (is_log_foi) {
+    # Normal distribution defaults
+    foi_defaults <- c(
+      foi_defaults,
+      list(
+        foi_sigma_rw_loc = prior_default$mean,
+        foi_sigma_rw_sc = prior_default$sd
+      )
+    )
+  } else {
+    # Cauchy distribution defaults
+    foi_defaults <- c(
+      foi_defaults,
+      list(
+        foi_sigma_rw_loc = prior_default$location,
+        foi_sigma_rw_sc = prior_default$scale
+      )
+    )
+  }
+  stan_data <- c(
+    stan_data,
+    foi_defaults
+  )
+
+  if (is_seroreversion) {
+    seroreversion_defaults <- list(
+      seroreversion_prior_index = prior_default$index,
+      seroreversion_min = prior_default$min,
+      seroreversion_max = prior_default$max,
+      seroreversion_mean = prior_default$mean,
+      seroreversion_sd = prior_default$sd
+    )
+    stan_data <- c(
+      stan_data,
+      seroreversion_defaults
+    )
+  }
+
+  return(stan_data)
+}
+
+#' Builds stan data for sampling depending on the selected model
+#'
+#' @inheritParams fit_seromodel
+#' @return List with necessary data for sampling the specified model
+#' @export
+build_stan_data <- function(
+    serosurvey,
+    model_type = "constant",
+    foi_prior = sf_uniform(),
```

Comment:
I checked the config.yml file and it did seem to have a max default of 10 (which I think is fine); I just wanted to double-check that I read it correctly?

Reply:
Yes, that's right. The default uniform distribution is `sf_uniform(0, 10)` and for the normal distribution it is `sf_normal(0, 1)`.